### ISSUE TITLE
Custom restricted access

### ISSUE THREAD

**Issue by BottlecapDave** [PTST/O365Calendar-HomeAssistant#102]
Hi there. First off, thanks for creating this component.

One thing however I'm a little uncomfortable with is the amount of scopes that are needed to get this component to work. For instance, I'm personally just needing read only access on calendars.

It would be really nice if you could turn on/off parts of this component (e.g. calendar_read, calendar_write) which would build up the required scopes. This way users could give as much or as little access as they want to the component.

**Comment by BottlecapDave** [comment]
I've created a merge request to support this functionality

**Comment by saicrazyfire** [comment]
I agree 100%. All I care about are the Calendar read scopes. This will also make it easier to get my app installed in my corporate environment
